### DISCUSSION TITLE
Push docker image after integration tests

### DISCUSSION THREAD

**Discussion by heineson** [quarkusio/quarkus#30395] [Q&A] [answered]
Hi!

I am using `quarkus-container-image-jib` to build docker images, and push them to the repo by setting the property `quarkus.container-image.push=true`. So far, so good.
However, this makes the image push happen as soon as the image is built during the package phase in Maven. I want to delay pushing until after integration tests have completed successfully, if this is possible? Building the image must still be done during the package phase.

Thanks for any advice!

[a comment by an automated account is not shown]

**Comment by geoand** [comment, marked as answer]
This is not something we support yet, but we do have it on the roadmap

**Comment by dasteg** [comment]
there is "quarkus.native.reuse-existing=true" which should prevent creating new native binaries if there is already one present. if you don't do "mvn clean" between your steps you should be fine.

**Comment by telapo** [comment]
This is a feature I am interested in...just a kind push

**Comment by querqueq** [comment]
Shouldn't pushing to a container registry happen in the maven deploy phase? Seems counterintuitive to me that it happens during the package phase.

**Comment by PhilippParis** [comment]
Hi!
is there any news regarding that issue?

**Reply by geoand** [reply]
Not yet

**Comment by cfitzw** [comment]
I too am interested in this issue....

We are trying to:
1. Package and Build (no push) --> works fine
    ```bash
    ./mvnw package --batch-mode --no-transfer-progress \
              -Dquarkus.container-image.build=true \
              -Dquarkus.container-image.push=false \
              -Dquarkus.container-image.builder=jib ...
    ```
2. Scan the container with external tooling --> works fine
3. Assuming no threats, push the previously build --> however, it seems the push still is forcing a rebuild.
    ` ./mvnw quarkus:image-push`

We **ONLY** want to perform a push.
